File: core/core-domain/src/main/java/com/imstargg/core/domain/brawlstars/Brawler.java
package com.imstargg.core.domain.brawlstars;

import com.imstargg.core.domain.BrawlStarsId;
import com.imstargg.core.domain.MessageCollection;
import com.imstargg.core.enums.BrawlerRarity;
import com.imstargg.core.enums.BrawlerRole;
import jakarta.annotation.Nullable;

import java.util.List;

public record Brawler(
        BrawlStarsId id,
        MessageCollection names,
        BrawlerRarity rarity,
        BrawlerRole role,
        List<Gadget> gadgets,
        List<Gear> gears,
        List<StarPower> starPowers,
        @Nullable String imagePath
) {

}

File: frontend/imstargg-frontend/src/model/Brawler.ts
import {BrawlerRarity} from "@/model/BrawlerRarity";
import {BrawlerRole} from "@/model/BrawlerRole";
import Gadget from "@/model/Gadget";
import Gear from "@/model/Gear";
import StarPower from "@/model/StarPower";

export default interface Brawler {
  id: number;
  name: string;
  rarity: BrawlerRarity;
  role: BrawlerRole;
  gadgets: Gadget[];
  gears: Gear[];
  starPowers: StarPower[];
}